magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/ActivityLifecycleAdapter.kt

```diff
@@ -1,13 +1,19 @@
 package com.wealthfront.magellan.lifecycle
 
 import android.app.Activity
+import android.content.Context
 import androidx.activity.ComponentActivity
 import androidx.annotation.LayoutRes
 import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.Lifecycle as ActivityLifecycle
+import androidx.lifecycle.LifecycleOwner as ActivityLifecycleOwner
 import com.wealthfront.magellan.R
 import com.wealthfront.magellan.ScreenContainer
 import com.wealthfront.magellan.core.Navigable
-import androidx.lifecycle.LifecycleOwner as ActivityLifecycleOwner
+import com.wealthfront.magellan.lifecycle.LifecycleState.Created
+import com.wealthfront.magellan.lifecycle.LifecycleState.Destroyed
+
+private var adapterMap = emptyMap<Navigable, Pair<ActivityLifecycleAdapter, ActivityLifecycle>>()
```

Comment:
nit: rename to ActivityLifecycleMap or something like that. Feel free to do in a separate PR as well or when you get to the java doc.